FinnUI/Sources/Components/SearchFilterTags/SearchFilterTagsView.swift

```diff
@@ -0,0 +1,255 @@
+    public static let height = 2 * SearchFilterTagsView.verticalMargin + SearchFilterTagCell.height
+
+    // MARK: - Internal properties
+    static let font = UIFont.captionStrong
+
+    // MARK: - Private properties
+
+    private static let verticalMargin = .spacingS + .spacingXXS
+    private static let horizontalMargin: CGFloat = .spacingS
+    private static let cellSpacing: CGFloat = .spacingXS
+    private static let filterButtonTrailingMargin: CGFloat = .spacingS
+
+    private lazy var collectionViewLayout: UICollectionViewLayout = {
+        let layout = UICollectionViewFlowLayout()
+        layout.scrollDirection = .horizontal
+        layout.minimumLineSpacing = .spacingS
+        layout.minimumInteritemSpacing = SearchFilterTagsView.cellSpacing
+        return layout
+    }()
+
+    private lazy var collectionView: UICollectionView = {
+        let collectionView = UICollectionView(frame: .zero, collectionViewLayout: collectionViewLayout)
+        collectionView.translatesAutoresizingMaskIntoConstraints = false
+        collectionView.backgroundColor = .clear
+        collectionView.showsHorizontalScrollIndicator = false
+        collectionView.dataSource = self
+        collectionView.delegate = self
+        collectionView.alwaysBounceHorizontal = true
+        collectionView.register(SearchFilterTagCell.self)
+        collectionView.contentInset = UIEdgeInsets(
+            trailing: SearchFilterTagsView.horizontalMargin
+        )
+        return collectionView
+    }()
+
+    private lazy var filterButtonView: SearchFilterButtonView = {
+        let button = SearchFilterButtonView(title: viewModel.filterButtonTitle, icon: viewModel.filterIcon)
+        button.translatesAutoresizingMaskIntoConstraints = false
+        button.delegate = self
+        return button
+    }()
+
+    private lazy var separatorView: UIView = {
+        let view = UIView(withAutoLayout: true)
+        view.backgroundColor = .tableViewSeparator
+        return view
+    }()
+
+    private let viewModel: SearchFilterTagsViewModel
+    private var searchFilterTags = [SearchFilterTagCellViewModel]()
+
+    private var maxFilterButtonWidth: CGFloat {
+        filterButtonView.contentWidth
+    }
+
+    private var minFilterButtonWidth: CGFloat {
+        SearchFilterButtonView.minWidth
+    }
+
+    private var filterButtonWidthConstraint: NSLayoutConstraint!
+
+    // MARK: - Init
+
+    public init(viewModel: SearchFilterTagsViewModel) {
+        self.viewModel = viewModel
+        super.init(frame: .zero)
+        setup()
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    // MARK: - Setup
+
+    private func setup() {
+        backgroundColor = .bgPrimary
+
+        addSubview(filterButtonView)
+        addSubview(collectionView)
+        addSubview(separatorView)
+
+        filterButtonWidthConstraint = filterButtonView.widthAnchor.constraint(equalToConstant: maxFilterButtonWidth)
```

Comment:
Could this be placed as `private lazy var filterButtonWidthConstraint = filterButtonView.widthAnchor.constraint(equalToConstant: maxFilterButtonWidth)` instead?